akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala

```diff
@@ -972,13 +972,61 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
    *
    * '''Cancels when''' downstream cancels
    *
-   * @param seed Provides the first state for extrapolation using the first unconsumed element
    * @param extrapolate Takes the current extrapolation state to produce an output element and the next extrapolation
    *                    state.
    */
   def expand[U](extrapolate: function.Function[Out, java.util.Iterator[U]]): SubSource[U, Mat] =
     new SubSource(delegate.expand(in ⇒ extrapolate(in).asScala))
 
+  /**
+   * Allows a faster downstream to progress of a slower upstream.
+   *
+   * This is achieved by introducing "extrapolated" elements - based on those from upstream - whenever downstream
+   * signals demand.
+   *
+   * Pressurize does not support [[akka.stream.Supervision#restart]] and [[akka.stream.Supervision#resume]].
+   * Exceptions from the `extrapolate` function will complete the stream with failure.
+   *
+   * '''Emits when''' downstream stops backpressuring, AND EITHER upstream emits OR initial element is present OR
+   * `extrapolate` is non-empty and applicable
+   *
+   * '''Backpressures when''' downstream backpressures or current `extrapolate` runs empty
+   *
+   * '''Completes when''' upstream completes and current `extrapolate` runs empty
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @param extrapolate takes the current upstream element and provides a sequence of "extrapolated" elements based
+   *                    on the original, to be emitted in case downstream signals demand
+   */
+  def pressurize[U >: Out](extrapolate: function.Function[U, java.util.Iterator[U]]): SubSource[U, Mat] =
```

Comment:
This will not work for Java (in fact I think we just realized not even the upper bounds really work like we have them). I think we'll have to limit it to `Function[Out, Iterator[Out]]` for Java.

Use `:javap f.q.c.Name` to verify in the sbt or scala console

Reply:
Sorry, I'm not sure I understand (especially given the remarks below). What solution do you suggest and what should I verify in the bytecode?